### PR TITLE
[Bugfix][V1] triton_unified_attention: apply tile_mask in USE_TD KV load

### LINKED ISSUE #43428: [Bug]: Endless '!' output when a long context request is sent on qwen3.5/3.6, B60 gpus
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Collecting environment information...
==============================
        System Info
==============================
OS                           : Ubuntu 26.04 LTS (x86_64)
GCC version                  : (Ubuntu 15.2.0-16ubuntu1) 15.2.0
Clang version                : 21.1.8 (6ubuntu1)
CMake version                : version 4.3.1
Libc version                 : glibc-2.43

==============================
       PyTorch Info
==============================
PyTorch version              : 2.11.0+xpu
Is debug build               : False
CUDA used to build PyTorch   : None
ROCM used to build PyTorch   : N/A
XPU used to build PyTorch    : 20250302

==============================
      Python Environment
==============================
Python version               : 3.12.13 | packaged by Anaconda, Inc. | (main, Mar 19 2026, 20:20:58) [GCC 14.3.0] (64-bit runtime)
Python platform              : Linux-6.17.0-29-generic-x86_64-with-glibc2.43

==============================
      Intel XPU / GPU Info
==============================
Is XPU available             : True
XPU runtime version          : 20250302
Intel GPU models             :
GPU 0: Intel(R) Arc(TM) Pro B60 Graphics
GPU 1: Intel(R) Arc(TM) Pro B60 Graphics
GPU 2: Intel(R) Arc(TM) Pro B60 Graphics
GPU 3: Intel(R) Arc(TM) Pro B60 Graphics


--Compile time--
oneAPI compiler version      : 2025.3.2
SYCL compiler build          : 2025.3.2.20260112
oneCCL version               : 1.0

--Runtime--
Intel Graphics Compiler (IGC): 2.34.4-1260~26.04
Intel GMM (libigdgmm)        : 22.10.0-1~26.04~ppa1
Level Zero loader version    : 1.28.2-2
Level Zero driver version    : 26.18.38308.1-1~26.04~ppa1
vLLM XPU kernels version     : 0.1.8

==============================
          CPU Info
==============================
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  48
On-line CPU(s) list:                     0-47
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) w7-3455
CPU family:                              6
Model:                                   143
Thread(s) per core:                      2
Core(s) per socket:                      24
Socket(s):                               1
Stepping:                                5
CPU(s) scaling MHz:                      23%
CPU max MHz:                             4800.0000
CPU min MHz:                             800.0000
BogoMIPS:                                4992.00
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect user_shstk avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                          VT-x
L1d cache:                               1.1 MiB (24 instances)
L1i cache:                               768 KiB (24 instances)
L2 cache:                                48 MiB (24 instances)
L3 cache:                                67.5 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-47
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Vulnerable
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

==============================
Versions of relevant libraries
==============================
[pip3] numpy==2.3.5
[pip3] pyzmq==27.1.0
[pip3] torch==2.11.0+xpu
[pip3] torchao==0.17.0+xpu
[pip3] torchaudio==2.11.0+xpu
[pip3] torchvision==0.26.0+xpu
[pip3] transformers==5.8.1
[pip3] triton==3.6.0
[pip3] triton-xpu==3.7.0
[conda] mkl                                         2025.3.1                         pypi_0           pypi
[conda] numpy                                       2.3.5                            pypi_0           pypi
[conda] onemkl-license                              2025.3.1                         pypi_0           pypi
[conda] onemkl-sycl-blas                            2025.3.1                         pypi_0           pypi
[conda] onemkl-sycl-dft                             2025.3.1                         pypi_0           pypi
[conda] onemkl-sycl-lapack                          2025.3.1                         pypi_0           pypi
[conda] onemkl-sycl-rng                             2025.3.1                         pypi_0           pypi
[conda] onemkl-sycl-sparse                          2025.3.1                         pypi_0           pypi
[conda] pyzmq                                       27.1.0                           pypi_0           pypi
[conda] torch                                       2.11.0+xpu                       pypi_0           pypi
[conda] torchao                                     0.17.0+xpu                       pypi_0           pypi
[conda] torchaudio                                  2.11.0+xpu                       pypi_0           pypi
[conda] torchvision                                 0.26.0+xpu                       pypi_0           pypi
[conda] transformers                                5.8.1                            pypi_0           pypi
[conda] triton                                      3.6.0                            pypi_0           pypi
[conda] triton-xpu                                  3.7.0                            pypi_0           pypi

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
vLLM Version                 : 0.21.1rc1.dev214+g4658bf882 (git sha: 4658bf882)
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled; XPU: Enabled
GPU Topology:
  Could not collect

==============================
     Environment Variables
==============================
VLLM_ALLOW_LONG_MAX_MODEL_LEN=1
VLLM_WORKER_MULTIPROC_METHOD=spawn
OMP_NUM_THREADS=48
VLLM_XPU_ENABLE_XPU_GRAPH=1
LD_LIBRARY_PATH=/opt/intel/oneapi/2025.3/opt/mpi/libfabric/lib:/opt/intel/oneapi/2025.3/opt/compiler/lib:/opt/intel/oneapi/2025.3/lib
CCL_ROOT=/opt/intel/oneapi/2025.3
I_MPI_ROOT=/opt/intel/oneapi/2025.3
CCL_CONFIGURATION_PATH=
CCL_CONFIGURATION=cpu_gpu_dpcpp
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
TORCHINDUCTOR_CACHE_DIR=/tmp/torchinductor_cjai
```

</details>

### 🐛 Describe the bug

All vllm serve cases are affected except parallelism(not including dp) is disabled.
e.g 
```bash
vllm serve "/data/models/Qwen3.6-27B-GPTQ-Pro-4bit/"   --dtype=bfloat16     --port 8000      --gpu-memory-util 0.9  --served-model-name qwen3.6-27b --enable-auto-tool-choice --reasoning-parser qwen3 --tool-call-parser qwen3_xml   -cc.cudagraph_mode=full -tp=2 -dp=2

vllm serve "/data/models/Qwen3.6-27B"   --dtype=bfloat16     --port 8000      --gpu-memory-util 0.9  --served-model-name qwen3.6-27b --enable-auto-tool-choice --reasoning-parser qwen3 --tool-call-parser qwen3_xml   --enforce-eager -tp=4
```
are affected.
```bash
vllm serve "/data/models/Qwen3.5-9B/"   --dtype=bfloat16     --port 8000      --gpu-memory-util 0.9  --served-model-name qwen3.6-27b --enable-auto-tool-choice --reasoning-parser qwen3 --tool-call-parser qwen3_xml -dp=4 
```
This won't cause the problem.

When a small context request is sent, vllm works perfect. When a large context one is sent, it will cause ALL concurrent requests to generate infinite "!!!!!!!!!!!.....".

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by lcheng321**
## Purpose
Fixes #43428 (Endless "!" output on Intel XPU long-context requests).

**Root Cause & Fix:**
The `USE_TD` KV load path misses the `tile_mask` applied in the standard non-TD path. Consequently, uninitialized KV-cache memory in the sequence tail causes `0.0 * NaN = NaN` during `tl.dot(P, V)`, poisoning the logits and returning token 0 (`!`). 

This PR fixes it by explicitly applying `tile_mask` to `K` and `V` in the `USE_TD` branch, mirroring the safety logic of the non-TD path.

## Test Plan
1. **Numerical Proof:** A local PyTorch script verified that masking the Value buffer prevents the `0.0 * NaN = NaN` propagation.
2. **Hardware Verification:** I do not have Intel hardware to test the XPU path. Requested users in #43428 to verify end-to-end on real hardware.

### CODE REVIEW

**Comment by jikunshang** [comment]
maybe you can try with VLLM_TRITON_ATTN_USE_TD=1 on xpu.

**Comment by lcheng321** [comment]
> maybe you can try with VLLM_TRITON_ATTN_USE_TD=1 on xpu.

Hi @jikunshang 
Thanks for the suggestion! That’s a great way to verify the fix. I’ve updated the PR (#44850) to include this testing method.
